src/analyses/goto_check.cpp

```diff
@@ -1164,10 +1164,20 @@ void goto_checkt::pointer_validity_check(
 
   const exprt &pointer=expr.pointer();
 
-  auto size_of_expr_opt = size_of_expr(expr.type(), ns);
-  CHECK_RETURN(size_of_expr_opt.has_value());
+  exprt size;
 
-  auto conditions = address_check(pointer, size_of_expr_opt.value());
+  if(expr.type().id() == ID_empty)
+  {
+    size = from_integer(1, size_type());
```

Comment:
Should we change `size_of_expr` to return `1` here instead?

Reply:
Not sure. There are quite a number of uses in the codebase that check that the return value of `size_of_expr()` is not null, and if it is throw an exception or have some other handling. Probably we should review those at some point, though that's out of scope of this PR.

Reply:
Is the purpose of this to create an address check that fails?

Reply:
No the address check should succeed. Basically `*p` with `p` being a pointer to void is valid as long as `p` points to valid memory, though then using the value would be undefined behavior.

Reply:
Please explain the magic `1` in a comment.